### ISSUE TITLE
chore: bump sha3 version

### ISSUE THREAD

**Issue by yihau** [anza-xyz/agave#647]
#### Problem

discussed with Sam earlier and we all agree that we should bump sha3 to the latest but we will need to update code base for some API breaking changes 😢

**Comment by Xuanwo** [comment]
Hi, @samkim-crypto, are you still on this issue? I'm interested to tackle this issue.

**Comment by samkim-crypto** [comment]
Yeah, it seems like to update sha3, we would need to bump curve25519-dalek to v4, which is on hold at the moment due to some breaking changes. You are welcome to tackle it if this is not the case though!

**Comment by Xuanwo** [comment]
> Yeah, it seems like to update sha3, we would need to bump curve25519-dalek to v4

Got it, I will take a look.

**Comment by Xuanwo** [comment]
Hi, @samkim-crypto, I have submitted a PR https://github.com/anza-xyz/agave/pull/946 for this. Welcome to take a review, thanks!
